### ISSUE TITLE
move_unit broken

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#5877]
### Game and System Information
 Battle for Wesnoth v1.15.13+dev (38241beac46-Modified) x86_64

### Describe the bug
It looks like the MOVE_UNIT macro is broken (possibly the issue is with [move_unit]).  It is used in mainline, so if this is confirmed, it is a blocker.

```Checking lua scripts... ok
20210619 19:41:32 error scripting/lua: lua/wml/move_unit.lua:88: attempt to call a nil value (field 'find_vacant_hex')
stack traceback:
        lua/wml/move_unit.lua:88: in local 'cmd'
        lua/wml-utils.lua:144: in field 'handle_event_commands'
        lua/wml-flow.lua:5: in function <lua/wml-flow.lua:4>
```

### To Reproduce
Load a save file that uses MOVE_UNIT, for example "UtBS- Speaking with the Fishes"

**Comment by stevecotton** [comment]
Argh there's a load of uses of level 1 deprecated APIs that are still in mainline, are called from the unit tests, yet somehow aren't triggering warnings during the unit tests.

PR #5863 put level 1 deprecation on `wesnoth.find_vacant_tile`, moving it to `wesnoth.paths.find_vacant_tile`, without changing the calling places (see 7f162cbd24c75ce65831013f467845b21a8976f6 for a list of the callers).

Now that I see this issue, I'm also seeing that the other deprecations in #5863 were missing changes in mainline. For example, there's still a lot of `wesnoth.find_reach` calls.

**Comment by doofus-01** [comment]
So, the fix is in the Lua API, not the wml tag or macro, is that right?

**Comment by Pentarctagon** [comment]
Fixed in 7b383cee6859d2d08074c1d7fd8db349c6441812

**Comment by Pentarctagon** [comment]
> Argh there's a load of uses of level 1 deprecated APIs that are still in mainline, are called from the unit tests, yet somehow aren't triggering warnings during the unit tests.

It looks like the CI doesn't provide the `-d` argument when running the tests.

**Comment by stevecotton** [comment]
> Fixed in [7b383ce](https://github.com/wesnoth/wesnoth/commit/7b383cee6859d2d08074c1d7fd8db349c6441812)

That's not right, or at least it'll break UMC.

* `wesnoth.find_vacant_tile` was part of the Lua API in 1.14, so still needs a deprecation wrapper.
* `wesnoth.paths.find_vacant_hex` is the non-deprecated name for it.
* `wesnoth.find_vacant_hex` is the result of a search-and-replace, it doesn't need deprecation because 24 hours ago nothing was trying to call it.

**Comment by CelticMinstrel** [comment]
I updated deprecated references to find_vacant_tile, find_reach, and find_vision_range. Let me know if you spot any others.
